saleor/checkout/complete_checkout.py

```diff
@@ -665,7 +665,13 @@ def _prepare_checkout_with_transactions(
 ):
     """Prepare checkout object with transactions to complete the checkout process."""
     clean_billing_address(checkout_info, CheckoutErrorCode)
-    if checkout_info.checkout.authorize_status != CheckoutAuthorizeStatus.FULL:
+    if (
+        checkout_info.checkout.authorize_status != CheckoutAuthorizeStatus.FULL
+        and not (
```

Comment:
We shouldn't have a condition like `checkout_info.checkout.authorize_status == CheckoutAuthorizeStatus.NONE` here. 
You may want to have an order with two payments, first payment could be processed before creating the order, next one could be added later.
Similar case, when you require to cover 30% of the order total  for pre-orders, etc. 
I am saying that if `allow_to_create_order_without_payment` is set to True, we shouldn't validate any payment's satuses